server/server/src/main/kotlin/org/jetbrains/bsp/bazel/server/sync/BspProjectMapper.kt

```diff
@@ -600,4 +579,36 @@ class BspProjectMapper(
 
     return toRustWorkspaceResult(requestedModules, allRustModules)
   }
+
+  companion object {
+    @JvmStatic
+    fun buildDependencyModulesStatic(project: Project, params: DependencyModulesParams): DependencyModulesResult {
+      val targetSet = params.targets.toSet()
+      val cache = mutableMapOf<String, List<DependencyModule>>()
```

Comment:
I think this cache can be an instance field of `BazelProjectMapper`, then you won't need the static function

Reply:
Other than this the change looks good

Reply:
I made it a static function because I couldn't figure out how to create a BspProjectMapper instance (had to fake out a bunch of other fields). Any pointers there would be appreciated.

Reply:
Hang on - does this being a static function mean that `cache` will be shared across all function calls? That was not my intention.

(This is my first foray into Kotlin)

Reply:
Sorry, I just overlooked this, your code makes sense actually

Reply:
I'm gonna fix the failing test in a second